lib/ch/query.ex

```diff
@@ -236,7 +236,11 @@ defimpl DBConnection.Query, for: Ch.Query do
   defp encode_map_params([] = empty), do: empty
 
   defp encode_array_param(s) when is_binary(s) do
-    [?', to_iodata(s, 0, s, []), ?']
+    [?', escape_array_param(s), ?']
+  end
+
+  defp encode_array_param(%s{} = param) when s in [Date, NaiveDateTime] do
+    [?', encode_param(param), ?']
```

Comment:
I think the set of text values is smaller than the set of non-test values in query params. So it's easier to have special clauses for text values (binaries, dates, naive datetimes) and `encode_param` the rest.

Reply:
I like this approach way more